src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -56,11 +57,11 @@ impl BDA {
 
     /// Load a BDA on initial setup of a device.
     /// Returns None if no BDA appears to exist.
-    pub fn load<F>(f: &mut F) -> StratisResult<Option<BDA>>
+    pub fn load<F>(f: &mut F, index: usize) -> StratisResult<Option<BDA>>
     where
         F: Read + Seek + SyncAll,
     {
-        let header = match StaticHeader::setup(f)? {
+        let header = match StaticHeader::setup(f, index)? {
```

Comment:
If the static header is offset, then I think that the MDA header below will need to be offset as well.